apple/internal/rule_factory.bzl

```diff
@@ -525,6 +525,13 @@ use only extension-safe APIs.
 A list of `.h` files that will be publicly exposed by this framework. These headers should have
 framework-relative imports, and if non-empty, an umbrella header named `%{bundle_name}.h` will also
 be generated that imports all of the headers listed here.
+""",
+            ),
+            "custom_umbrella_header": attr.label(
```

Comment:
I have no preference on the name here, this could also just be `umbrella_header` if folks wanted that